activesupport/lib/active_support/concern.rb

```diff
@@ -104,6 +104,13 @@ def initialize
       end
     end
 
+    class UnreachableIncludedBlock < StandardError #:nodoc:
+      def initialize
+        super "This 'included' block will never be run because the Concern was already included. " +
+          "Check your code for circular references from the Concern to the including class."
```

Comment:
Again, I don't think this is a clear error message at all. It should be something like "The `included` block for #{concern_name} will never be run. This is usually caused by referencing a class by name in the `included` block, and then including the module into that class. To work around this issue, override `self.included` explicitly on #{concern_name}, rather than using `ActiveSupport::Concern`"